### DISCUSSION TITLE
Automatic Background Updates?

### DISCUSSION THREAD

**Discussion by chrisprabhu** [aonez/Keka#1041] [Ideas]
I love Keka but almost every single time that I use it I have to update and restart it. Is there any way you can make it update automatically in the background before I open it?

**Comment by gingerbeardman** [comment]
Uncheck *Preferences > General > Automatically check for updates*.

Then you can manage updates manually. 
If it does what you need there is no pressing reason to update. 
Though it's still worth updating occasionally, for security fixes amongst other things.

**Comment by gingerbeardman** [comment]
But, back to your point, I wonder if it could update silently on Quit (like many other apps) rather than requiring a restart? @aonez 

That way next time you launch it will be pretty much up-to-date, and your current unarchiving task will not be interrupted.

**Comment by aonez** [comment]
You can set Keka to automatically download and install new versions:

<img width="732" alt="Screenshot 2022-02-07 at 16 39 55" src="https://user-images.githubusercontent.com/480163/152821574-f7e6b839-c026-4288-823a-2f26b097cfec.png">

Also using the Terminal.app:

```
defaults write com.aone.keka SUAutomaticallyUpdate 1
defaults write com.aone.keka SUEnableAutomaticChecks 1
```

It will download in the background and install once Keka quits. @gingerbeardman this feature is working properly now, let me know if it is not for you.

Maybe I could add that as an option in the main Keka preferences.

**Comment by gingerbeardman** [comment]
I've just switched this option on.

For others: https://sparkle-project.org/documentation/customization/

`SUEnableAutomaticChecks`
**Setting to YES enables checking for updates (but not installation) by default, without asking your users for permission first.** By default, if it’s not set to any value, users will be prompted for permission before the first update check. Setting to NO disables update checks, but can be overridden by setting automaticallyChecksForUpdates.

`SUAutomaticallyUpdate`
Default: NO. **Enables automatic download and installation of updates. If set to YES, Sparkle will attempt to download and install new updates silently in the background.**
